Source/CombatExtended/CombatExtended/StatParts/Base/StatPart_StatSelect.cs

```diff
@@ -41,22 +45,49 @@ public override void TransformValue(StatRequest req, ref float result)
             {
                 if (apparelStat != null && pawn.apparel != null)
                 {
-                    ThingOwner<Apparel> wornApparel = pawn.apparel.wornApparel;
+                    if (sumApparelsStat)
+                    {
+                        float value = 0;
+                        ThingOwner<Apparel> wornApparel = pawn.apparel.wornApparel;
+
+                        for (int i = 0; i < wornApparel.Count; i++)
+                            value += GetEquipmentStat(wornApparel[i], apparelStat);
 
-                    for (int i = 0; i < wornApparel.Count; i++)
-                        result = Select(result, GetEquipmentStat(wornApparel[i], apparelStat));
+                        result = Select(value, result);
+                    }
+                    else
+                    {
+                        ThingOwner<Apparel> wornApparel = pawn.apparel.wornApparel;
+
+                        for (int i = 0; i < wornApparel.Count; i++)
+                            result = Select(result, GetEquipmentStat(wornApparel[i], apparelStat));
+                    }
                 }
                 if (weaponStat != null && pawn.equipment != null && pawn.equipment.Primary != null)
                 {
                     result = Select(result, GetEquipmentStat(pawn.equipment.Primary, weaponStat));
                 }
                 if (implantStat != null)
                 {
-                    List<Hediff> hediffs = pawn.health.hediffSet.hediffs;
-                    for (int i = 0; i < hediffs.Count; i++)
+                    if (sumImplantsStat)
+                    {
+                        List<Hediff> hediffs = pawn.health.hediffSet.hediffs;
+                        float value = 0f;
+                        for (int i = 0; i < hediffs.Count; i++)
+                        {
+                            if (hediffs[i] is Hediff_AddedPart addedpart && addedpart != null && addedpart.def.spawnThingOnRemoved != null)
+                                value += GetAddedPartStat(addedpart);
+                        }
+                        result = Select(value, result);
+                    }
+                    else
                     {
-                        if (hediffs[i] is Hediff_AddedPart addedpart && addedpart != null && addedpart.def.spawnThingOnRemoved != null)
-                            result = Select(result, GetAddedPartStat(addedpart));
+                        List<Hediff> hediffs = pawn.health.hediffSet.hediffs;
+                        for (int i = 0; i < hediffs.Count; i++)
+                        {
+                            if (hediffs[i] is Hediff_AddedPart addedpart && addedpart != null && addedpart.def.spawnThingOnRemoved != null)
```

Comment:
The `addedpart != null` check shouldn't be needed.  `hediffs[i]` should never be `null` (and if it were, it would not return `true` on `is Hediff_AddedPart`), so if it *is* a `Hediff_AddedPart` it won't be null.